### PR TITLE
test: add comprehensive test coverage for LP page, components and hooks

### PR DESCRIPTION
**Pull request by dicethedev**
# ✅ LP Page and Components Test Coverage

This update introduces a comprehensive testing suite for LP (Liquidity Provider) components and functionality, ensuring all major UI states, user interactions, and data flows are covered and reliable. The goal is to prevent regressions, improve developer confidence, and support future iterations on the LP experience.

### Components
✅ **LPPage Component**
- Connected vs disconnected wallet states
- LP vs non-LP user scenarios  
- Active request blocking logic
- Price change styling (positive/negative)
- Pool status indicators
- Responsive design classes

✅ **TradingViewWidget Component**
- Script creation and configuration
- Symbol parameter handling
- Component lifecycle and cleanup
- Container reference management

✅ **UnconnectedActionsCard Component**
- Wallet connection UI
- RainbowKit integration
- Styling and layout verification

### Hooks & Logic
✅ **useLPData Hook**
- Subgraph data fetching
- Loading and error states
- LP status determination
- Request blocking calculations

✅ **Utility Functions**
- US market hours detection
- Rebalance state calculations
- Time-based logic testing

### User States
🌀 Loading state for LP data  
❌ Empty state (no LP positions/requests)  
⚠️ Error state (network failures, subgraph issues)  
🔒 Blocked request states (current/previous cycle)  
💰 Active LP with positions and requests  
🚫 Non-LP users (limited functionality)

### Edge Cases
🔢 Large number formatting  
⚡ Zero/negative price changes  
🎯 Missing or invalid pool parameters  
🧹 Component cleanup and memory leaks

### External Dependencies  
🧱 Mocked wagmi hooks (useAccount, useReadContract)  
📊 Mocked subgraph queries  
🎨 Mocked UI components for isolation  
🌐 Mocked external widgets (TradingView)

## 🧼 Improvements

- **Increased reliability** of LP functionality
- **Prevents unhandled edge cases** during runtime
- **Establishes baseline** for future test expansion  
- **Removes flaky behavior** caused by external dependencies
- **Improves developer confidence** when making changes
- **Ensures consistent UX** across different user states

## 🚀 How to Run Tests

```bash
# Run all LP component tests
npm test -- --testPathPatterns="LP"

# Run specific test files
npm test lpPage.test.tsx
npm test lp-user.test.tsx
npm test tradingViewComponent.test.tsx
npm test UnconnectedActionsCard.test.tsx
npm test lp.test.ts
npm test lp.utils.test.ts

# Run with coverage report
npm test -- --coverage --testPathPatterns="LP"

# Watch mode for development
npm test -- --watch LP

### CODE REVIEW

File: src/hooks/lp.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export const isUSMarketOpen = (): boolean => {
 
 // Helper function to determine rebalance state based on pool data
 export const calculateRebalanceState = (
-  pool: Pool
+  pool?: Pool
```
Comment (bhargavaparoksham) [inline src/hooks/lp.ts:132]: We don't need to make this optional

**Review by bhargavaparoksham — APPROVED** [review]
